File: entities/doctor.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type DoctorStatus string

const (
	DoctorStatusNotSubmitted       DoctorStatus = "NOT_SUBMITTED"
	DoctorStatusPending            DoctorStatus = "PENDING"
	DoctorStatusAutoVerified       DoctorStatus = "AUTO_VERIFIED"
	DoctorStatusManualReview       DoctorStatus = "MANUAL_REVIEW"
	DoctorStatusVerified           DoctorStatus = "VERIFIED"
	DoctorStatusRejected           DoctorStatus = "REJECTED"
	DoctorStatusSuspended          DoctorStatus = "SUSPENDED"
	DoctorStatusDocumentsRequested DoctorStatus = "DOCUMENTS_REQUESTED"
)

type Doctor struct {
	ID                  uuid.UUID      `gorm:"primaryKey" json:"id"`
	PublicID            uuid.UUID      `gorm:"uniqueIndex;not null" json:"public_id"`
	FirstName           string         `gorm:"size:100;not null" json:"first_name"`
	LastName            string         `gorm:"size:100;not null" json:"last_name"`
	Gender              string         `gorm:"size:20" json:"gender"`
	DOB                 *time.Time     `json:"dob"`
	Mobile              string         `gorm:"size:20;uniqueIndex;not null" json:"mobile"`
	Email               string         `gorm:"size:255;uniqueIndex;not null" json:"email"`
	PasswordHash        string         `gorm:"size:255;not null" json:"-"`
	ProfilePhoto        string         `gorm:"type:text" json:"profile_photo,omitempty"`
	Languages           string         `gorm:"size:255" json:"languages,omitempty"`
	Biography           string         `gorm:"type:text" json:"biography,omitempty"`
	Status              DoctorStatus   `gorm:"size:50;default:'NOT_SUBMITTED';not null;index" json:"status"`
	FraudScore          int            `gorm:"default:0" json:"fraud_score"`
	MobileVerified      bool           `gorm:"default:false;not null" json:"mobile_verified"`
	EmailVerified       bool           `gorm:"default:false;not null" json:"email_verified"`
	LastLoginAt         *time.Time     `json:"last_login_at,omitempty"`
	FailedLoginAttempts int            `gorm:"default:0;not null" json:"failed_login_attempts"`
	AccountLockedUntil  *time.Time     `json:"account_locked_until,omitempty"`
	PasswordChangedAt   *time.Time     `json:"password_changed_at,omitempty"`
	AssignedAdminID     *uuid.UUID     `json:"assigned_admin_id,omitempty"`
	AssignedAt          *time.Time     `json:"assigned_at,omitempty"`
	PrescriptionEnabled bool           `gorm:"default:false;not null" json:"prescription_enabled"`
	CreatedAt           time.Time      `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt           time.Time      `gorm:"autoUpdateTime" json:"updated_at"`
	DeletedAt           gorm.DeletedAt `gorm:"index" json:"-"`
	DeletedBy           *uuid.UUID     `json:"deleted_by,omitempty"`

	// Relationships
	Licenses       []DoctorLicense       `gorm:"foreignKey:DoctorID;constraint:OnDelete:CASCADE" json:"licenses,omitempty"`
	Qualifications []DoctorQualification `gorm:"foreignKey:DoctorID;constraint:OnDelete:CASCADE" json:"qualifications,omitempty"`
	Clinics        []DoctorClinic        `gorm:"foreignKey:DoctorID;constraint:OnDelete:CASCADE" json:"clinics,omitempty"`
	Documents      []DoctorDocument      `gorm:"foreignKey:DoctorID;constraint:OnDelete:CASCADE" json:"documents,omitempty"`
	Histories      []VerificationHistory `gorm:"foreignKey:DoctorID;constraint:OnDelete:CASCADE" json:"histories,omitempty"`
	OTPs           []OTPVerification     `gorm:"foreignKey:DoctorID;constraint:OnDelete:CASCADE" json:"-"`
	RefreshTokens  []RefreshToken        `gorm:"foreignKey:DoctorID;constraint:OnDelete:CASCADE" json:"-"`
	Notes          []DoctorNote          `gorm:"foreignKey:DoctorID;constraint:OnDelete:CASCADE" json:"notes,omitempty"`
	Flags          []VerificationFlag    `gorm:"foreignKey:DoctorID;constraint:OnDelete:CASCADE" json:"flags,omitempty"`
	AdminActions   []AdminAction         `gorm:"foreignKey:DoctorID;constraint:OnDelete:CASCADE" json:"admin_actions,omitempty"`
}

func (d *Doctor) BeforeCreate(tx *gorm.DB) (err error) {
	if d.ID == uuid.Nil {
		d.ID = uuid.New()
	}
	if d.PublicID == uuid.Nil {
		d.PublicID = uuid.New()
	}
	return nil
}

File: python_backend/entities/doctor.py
import enum
import uuid
from sqlalchemy import Column, String, Integer, Boolean, DateTime, Enum, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from database import Base

class DoctorStatus(str, enum.Enum):
    NOT_SUBMITTED = "NOT_SUBMITTED"
    PENDING = "PENDING"
    AUTO_VERIFIED = "AUTO_VERIFIED"
    MANUAL_REVIEW = "MANUAL_REVIEW"
    VERIFIED = "VERIFIED"
    REJECTED = "REJECTED"
    SUSPENDED = "SUSPENDED"
    DOCUMENTS_REQUESTED = "DOCUMENTS_REQUESTED"

class Doctor(Base):
    __tablename__ = "doctors"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    public_id = Column(String(36), unique=True, index=True, nullable=False, default=lambda: str(uuid.uuid4()))
    first_name = Column(String(100), nullable=False)
    last_name = Column(String(100), nullable=False)
    gender = Column(String(20), nullable=True)
    dob = Column(DateTime(timezone=True), nullable=True)
    mobile = Column(String(20), unique=True, index=True, nullable=False)
    email = Column(String(255), unique=True, index=True, nullable=False)
    password_hash = Column(String(255), nullable=False)
    profile_photo = Column(Text, nullable=True)
    languages = Column(String(255), nullable=True)
    biography = Column(Text, nullable=True)
    status = Column(Enum(DoctorStatus, name="doctor_status_enum", native_enum=False), default=DoctorStatus.NOT_SUBMITTED, nullable=False, index=True)
    fraud_score = Column(Integer, default=0)
    mobile_verified = Column(Boolean, default=False, nullable=False)
    email_verified = Column(Boolean, default=False, nullable=False)
    last_login_at = Column(DateTime(timezone=True), nullable=True)
    failed_login_attempts = Column(Integer, default=0, nullable=False)
    account_locked_until = Column(DateTime(timezone=True), nullable=True)
    password_changed_at = Column(DateTime(timezone=True), nullable=True)
    assigned_admin_id = Column(String(36), nullable=True)
    assigned_at = Column(DateTime(timezone=True), nullable=True)
    prescription_enabled = Column(Boolean, default=False, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
    deleted_at = Column(DateTime(timezone=True), index=True, nullable=True)
    deleted_by = Column(String(36), nullable=True)

    # Relationships
    licenses = relationship("DoctorLicense", backref="doctor", cascade="all, delete")
    qualifications = relationship("DoctorQualification", backref="doctor", cascade="all, delete")
    clinics = relationship("DoctorClinic", backref="doctor", cascade="all, delete")
    documents = relationship("DoctorDocument", backref="doctor", cascade="all, delete")
    histories = relationship("VerificationHistory", backref="doctor", cascade="all, delete")
    otps = relationship("OTPVerification", backref="doctor", cascade="all, delete")
    refresh_tokens = relationship("RefreshToken", backref="doctor", cascade="all, delete")
    notes = relationship("DoctorNote", backref="doctor", cascade="all, delete")
    flags = relationship("VerificationFlag", backref="doctor", cascade="all, delete")
    admin_actions = relationship("AdminAction", backref="doctor", cascade="all, delete")
